includes/class-wc-payments-order-service.php

```diff
@@ -891,30 +891,39 @@ public function get_shipping_data_from_order( WC_Order $order ): array {
 
 	/**
 	 * Create the billing data array to send to Stripe when making a purchase, based on order's billing data.
+	 * It only returns the fields that are present in the billing section of the checkout.
 	 *
 	 * @param WC_Order $order The order that is being paid for.
 	 * @return array          The shipping data to send to Stripe.
 	 */
 	public function get_billing_data_from_order( WC_Order $order ): array {
-		return [
-			'name'    => implode(
-				' ',
-				array_filter(
-					[
-						$order->get_billing_first_name(),
-						$order->get_billing_last_name(),
-					]
-				)
-			),
-			'address' => [
-				'line1'       => $order->get_billing_address_1(),
-				'line2'       => $order->get_billing_address_2(),
-				'postal_code' => $order->get_billing_postcode(),
-				'city'        => $order->get_billing_city(),
-				'state'       => $order->get_billing_state(),
-				'country'     => $order->get_billing_country(),
-			],
+		$billing_fields       = array_keys( WC()->countries->get_address_fields( $order->get_billing_country() ) );
+		$address_field_to_key = [
+			'billing_city'      => 'city',
+			'billing_country'   => 'country',
+			'billing_address_1' => 'line1',
+			'billing_address_2' => 'line2',
+			'billing_postcode'  => 'postal_code',
+			'billing_state'     => 'state',
 		];
+		$field_to_key         = [
```

Comment:
Nitpick: since `$address_field_to_key` clearly indicates that it maps address fields, having this array name reflect its keys as well might potentially ease up the readability of this chunk of code, what do you think? 

```suggestion
		$contact_field_to_key         = [
```

Reply:
`$context_field_to_key` describes it better because the fields are related to contact, but I think that's just a coincidence. My reasoning was that these are fields placed in the first level of the array, and the other fields are placed under `'billing' => [ ... ] `.